server/socket-manager.js

```diff
@@ -10,11 +14,33 @@ const authenticateCookie = (socket, callback) => {
 };
 
 const socketManager = (socket) => {
-  socket.on('authenticate', () => {
-    authenticateCookie(socket, (err, data) => {
-      if(err) socket.emit('User not authenticated');
-      socket.emit('userdata', data);
-      // handles socket events
+  let user_id = '';
+
+  socket.on('authenticate', (_, clientCallback) => {
+    authenticateCookie(socket, (err, id) => {
+      if (err) {
+        socket.emit('authentication_error');
+        clientCallback('auth error');
+      }
+      getUserData(id, (data) => {
+        user_id = id;
+        clientCallback(null, id);
+        socket.emit('userdata', data);
+      });
+    });
+  });
+
+  socket.on('goal', (data, clientCallback) => {
```

Comment:
No!

You are registering this event listener even if the socket does not authenticate!! Only register this eventListener on successful authentication!

(I think)